File: car_decoration_app/lib/models/vehicle.dart
class Vehicle {
  final String id;
  final String brand;
  final String model;
  final int year;
  final String color;
  final String? plateNumber;
  final String mono;
  final bool isMain;

  const Vehicle({
    required this.id,
    required this.brand,
    required this.model,
    required this.year,
    required this.color,
    this.plateNumber,
    required this.mono,
    this.isMain = false,
  });

  String get displayName => '$brand $model';
  String get displayNameWithYear => '$brand $model $year';
}

File: api/CarDecoration.API/CarDecoration.API/Models/Vehicle.cs
namespace CarDecoration.API.Models;

public class Vehicle : BaseEntity
{
    public Guid OwnerId { get; set; }
    public string Brand { get; set; } = string.Empty;
    public string Model { get; set; } = string.Empty;
    public int Year { get; set; }
    public string? PlateNumber { get; set; }

    public User Owner { get; set; } = null!;
    public List<Request> Requests { get; set; } = [];
    public List<VehicleImage> VehicleImages { get; set; } = [];
}